pkg/diagnostics/diagnostics.go

```diff
@@ -49,10 +56,23 @@ func (r *Registry) RegisterConnectedPeersSource(
 	r.Registry.RegisterSource("connected_peers", func() string {
 		connectionManager := netProvider.ConnectionManager()
 		connectedPeers := connectionManager.ConnectedPeers()
+		connectedPeersAddrInfo := connectionManager.ConnectedPeersAddrInfo()
 
 		peersList := make([]map[string]interface{}, len(connectedPeers))
 		for i := 0; i < len(connectedPeers); i++ {
 			peer := connectedPeers[i]
+			peersAddrInfo := connectedPeersAddrInfo[i].Addrs
```

Comment:
Here we assume `len(connectedPeers) == len(peersAddrInfo)` and that the order of peers is the same.

I am pretty sure this is not always true. If one or more peers connect or disconnect after `connectionManager.ConnectedPeers()` and before `connectionManager.ConnectedPeersAddrInfo()`, the length of the slices will be different.

Also, there is nothing in libp2p `network.Dialer` interface documentation for `Peers() []peer.ID` saying that the returned peers are ordered. It could happen that elements of slices returned by `connectionManager.ConnectedPeersAddrInfo()` and `connectionManager.ConnectedPeers()` does not match.

I think we do not need to call `connectionManager.ConnectedPeers()` in this endpoint at all. Isn't `connectionManager.ConnectedPeersAddrInfo()` enough?

Reply:
You're right [cd25e92](https://github.com/keep-network/keep-core/pull/3255/commits/cd25e929b19592e13aba7e81c8c63017f5c003fd)